code/game/objects/items/stacks/stack.dm

```diff
@@ -463,6 +463,11 @@
 	target_stack.copy_evidences(src)
 	use(transfer, transfer = TRUE, check = FALSE)
 	target_stack.add(transfer)
+	if(target_stack.mats_per_unit != mats_per_unit)
+		var/list/temp_mats_list = list()
```

Comment:
We should doc this, just so people know why we're doing it. If you want me to give you some details on that feel free to bug me

Reply:
Sorry should have been more direct, I meant the temp_mats_list stuff primarily, but what you have now should be included too, though the why is more important then the what